File: core/src/main/java/com/spidey01/sxe/core/testing/NullConfiguration.java
/*-
 * Copyright (c) 2013-current, Terry Mathew Poulin <BigBoss1964@gmail.com>
 *
 * This software is provided 'as-is', without any express or implied warranty.
 * In no event will the authors be held liable for any damages arising from the
 * use of this software.
 *
 * Permission is granted to anyone to use this software for any purpose,
 * including commercial applications, and to alter it and redistribute it
 * freely, subject to the following restrictions:
 *
 *	1. The origin of this software must not be misrepresented; you must
 *	   not claim that you wrote the original software. If you use this
 *	   software in a product, an acknowledgment in the product
 *	   documentation would be appreciated but is not required.
 *
 *	2. Altered source versions must be plainly marked as such, and must
 *	   not be misrepresented as being the original software.
 *
 *	3. This notice may not be removed or altered from any source
 *	   distribution.
 */

package com.spidey01.sxe.core.testing;

import com.spidey01.sxe.core.Game;
import com.spidey01.sxe.core.GameEngine;
import com.spidey01.sxe.core.ResourceManager;
import com.spidey01.sxe.core.SceneManager;
import com.spidey01.sxe.core.cfg.Settings;
import com.spidey01.sxe.core.cfg.SettingsMap;
import com.spidey01.sxe.core.common.Subsystem;
import com.spidey01.sxe.core.io.SettingsFile;
import com.spidey01.sxe.core.logging.Log;
import com.spidey01.sxe.core.logging.Logging;
import com.spidey01.sxe.core.sys.Platform;

import java.io.File;

/** Utility class to setup a GameEngine configured with null input and display.
 */
public class NullConfiguration {
    private static final String TAG = "NullConfiguration";

    /** The defaults */
    public static GameEngine setup(Game game) {
        return setup(null, game);
    }


    public static GameEngine setup(String[] args, Game game) {
        return new GameEngine(
            new SettingsMap(args)
            , new NullDisplay(new NullOpenGL(), true)
            , new SceneManager()
            , game
            , new NullInputManager()
            , new ResourceManager()
            , new Logging()
            , null
            , new Platform(Platform.guess())
        );
    }
}

File: core/src/testing/NullConfiguration.cpp
/*-
 * Copyright (c) 2013-current, Terry Mathew Poulin <BigBoss1964@gmail.com>
 *
 * This software is provided 'as-is', without any express or implied warranty.
 * In no event will the authors be held liable for any damages arising from the
 * use of this software.
 *
 * Permission is granted to anyone to use this software for any purpose,
 * including commercial applications, and to alter it and redistribute it
 * freely, subject to the following restrictions:
 *
 *	1. The origin of this software must not be misrepresented; you must
 *	   not claim that you wrote the original software. If you use this
 *	   software in a product, an acknowledgment in the product
 *	   documentation would be appreciated but is not required.
 *
 *	2. Altered source versions must be plainly marked as such, and must
 *	   not be misrepresented as being the original software.
 *
 *	3. This notice may not be removed or altered from any source
 *	   distribution.
 */

#include "sxe/core/testing/NullConfiguration.hpp"

#include <sxe/core/config/SettingsMap.hpp>
#include <sxe/core/sys/Platform.hpp>
#include <sxe/core/testing/NullDisplay.hpp>

using std::make_shared;
using std::make_unique;
using sxe::core::config::Settings;
using sxe::core::config::SettingsMap;

namespace sxe { namespace core { namespace testing {

GameEngine::unique_ptr NullConfiguration::setup(Game::shared_ptr game)
{
    return setup(0, nullptr, game);
}


GameEngine::unique_ptr NullConfiguration::setup(int argc, char* argv[], Game::shared_ptr game)
{
    argc -= 1;
    argv += 1;

    return std::make_unique<sxe::core::GameEngine>
    (
        game
        , make_unique<SettingsMap>(argc, argv)
        , make_unique<NullDisplay>(true)
        , nullptr // scene manager
        , nullptr // input maanger -- TODO NullInputManager
        , nullptr // resource manager
        , nullptr // logging manager
        , nullptr // platform specific settings
        , sxe::core::sys::Platform()
    );
}

} } }

